soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -536,7 +536,7 @@ namespace Logic {
     Fish         = HasBottle && FishAccess;
     Fairy        = HasBottle && FairyAccess;
 
-    FoundBombchus   = (BombchuDrop || Bombchus || Bombchus5 || Bombchus10 || Bombchus20);
+    FoundBombchus   = (BombchuDrop || Bombchus || Bombchus5 || Bombchus10 || Bombchus20) && (BombBag || BombchusInLogic);
     CanPlayBowling  = (BombchusInLogic && FoundBombchus) || (!BombchusInLogic && BombBag);
     HasBombchus     = (BuyBombchus10 || BuyBombchus20 || (AmmoDrops.Is(AMMODROPS_BOMBCHU) && FoundBombchus));
```

Comment:
unless i'm misunderstanding this, it seems this would make it so logic never expects a scenario where someone
* has chus in logic off, has chu drops on
* finds chus
* uses chus

which is definitely possible because without chus in logic chus will drop if someone has chus (even if they don't have bomb bag)

based on
https://github.com/HarbourMasters/Shipwright/blob/1d7ad522225a46c2da614c6f96350e09658d75d4/soh/soh/Enhancements/randomizer/3drando/location_access.cpp#L79-L82

i'd think we want this to be

```suggestion
    FoundBombchus   = (BombchuDrop || Bombchus || Bombchus5 || Bombchus10 || Bombchus20);
    CanPlayBowling  = (BombchusInLogic && FoundBombchus) || (!BombchusInLogic && BombBag);
    HasBombchus     = (((BuyBombchus10 || BuyBombchus20) && (BombBag || BombchusInLogic)) || (AmmoDrops.Is(AMMODROPS_BOMBCHU) && FoundBombchus));
```

Reply:
The suggested line of code does nothing, as BuyBombchus is already accounting for bomb bag properly in the shop logic code. 

The purpose of that line is to patch the case where finding a bombchu chest was putting using bombchus in logic without a bomb bag on seeds where bombcus in logic is off, and while this matches the practicallity of what is happeneing in game, it is not in the spirit of disabling bombchus in logic and will create a scenario where HasBombchus is true when HasExplosives is false resulting in inconsistent logic.

Reply:
> HasBombchus is true when HasExplosives is false resulting in inconsistent logic

https://github.com/HarbourMasters/Shipwright/blob/1d7ad522225a46c2da614c6f96350e09658d75d4/soh/soh/Enhancements/randomizer/3drando/logic.cpp#L543

yes, that is intentional. chus in logic means you can be expected to use chus to get checks locked behind having explosives. chus out of logic means you won't be expected to get those checks using chus.

Reply:
It is not intended when "Bombchus in Logic" is off, which is what happens in this case. The full requirements are:
- "Bombchus in Logic" is off
- "Bombchu Drops" is on
- There is no bomb bag in logic
- There is a bombchu chest in logic
- The check looks for HasBombchus specifically, not HasExplosives

When these 5 things happen, the logic will see the bombchus as a valid item even when it shouldn't be because settings say bombchus should only be needed after the bomb bag. HasExplosives having that BombchusInLogic check and HasBombchus not having it is what causes the wierdness.